composeApp/src/commonMain/kotlin/io/github/couchtracker/db/common/defaultdata/DefaultDataHandler.kt

```diff
@@ -0,0 +1,54 @@
+package io.github.couchtracker.db.common.defaultdata
+
+import app.cash.sqldelight.Transacter
+
+/**
+ * Handles the insertion and upgrade of default data for a specific database.
+ */
+abstract class DefaultDataHandler<DB : Transacter>(
+    private val defaultData: DefaultData<DB>,
+) {
+
+    /**
+     * The latest version of default data that is handled.
+     */
+    protected abstract val latestVersion: Int
+
+    /**
+     * Handles creation on upgrade of the default data in the given [db].
+     *
+     * Every DB operation is inside a transaction, so that any failure will leave the DB untouched.
+     */
+    fun handle(db: DB) {
+        val latestVersion = latestVersion
+        db.transaction {
+            val currentVersion = db.getVersion()
+            if (currentVersion == null) {
+                defaultData.insert(db)
+                db.setVersion(latestVersion)
+            } else {
+                check(currentVersion <= latestVersion) {
```

Comment:
I don't think this should crash. An old app should be able to just open a new db, and just not insert any default data

Reply:
Fair enough